manifests/db.pp

```diff
@@ -31,7 +31,6 @@
     ensure        => $ensure,
     password_hash => mysql_password($password),
     provider      => 'mysql',
-    require       => Class['mysql::server'],
   }
```

Comment:
I would add here 
<code>Mysql_database[$dbname] -> Mysql_user["${user}@${host}"]</code>
or 
<code>require => Mysql_database[$dbname]</code>
this define is called only with all parameters anyway